### DISCUSSION TITLE
Collection Ordering Regressions

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#19537] [Q&A]
Version 2022.731.1
Collections are inconsistent in their ordering. This has two significant regressions since version 2022.723.0:
1. Making any change to a collection (adding, renaming, removing) causes the collections to reorder alphabetically. This may be intended, as this matches stable behavior, but this is a regression from lazer behavior.
2. The collection manager shows the manual order of collections. When right-clicking a map to apply collections, they show in the order created. When using the collections drop-down, they show alphabetically.

https://user-images.githubusercontent.com/74262407/182240751-1a2101df-b4b8-4d81-bc12-d6774e2460d1.mp4

When looking for test coverage for retaining the collection order, I found none. However, these behaviors can be observed in the `ManageCollectionsDialog` and `BeatmapCarousel` tests.

**Comment by peppy** [comment]
See the last bullet point on https://github.com/ppy/osu/pull/19430.

**Reply by Ernest314** [reply]
The linked PR says "I figure people will complain if they have been actually using the rearrangeability, else we can consider removing it"--is here the right place to "complain"? This functionality would be appreciated if it is implemented :)

**Comment by monochrome22** [comment]
In my opinion, all of the collection lists should be in order of creation, or if the user manually ordered them in Manage Collections, then in the order that they set. In stable, the listing is alphabetical, but at least for me, that ordering has always been frustrating to deal with; I've had to rename collections many times and use symbols just to get them in the right order.

If I understand correctly that the refreshing is caused by reordering, then this would also solve the issue that I pointed out at https://github.com/ppy/osu/pull/23741#issuecomment-1575491404.

If the ability to order alphabetically is wanted for the sake of matching stable behavior, then there should be a button or a checkbox for "Sort collections alphabetically". This should be a dangerous action. 
Post peppy reply edit: If not a dangerous action(which might get annoying if used often), then another option is to change the "Sort alphabetically" button to "Undo alphabetical sorting" after it's pressed, and if the user creates a new collection or manually reorders something, then it will change back to "Sort alphabetically".
What I'm thinking is: someone might have hundreds of collections which they have precisely ordered, and if they accidentally hit the sort alphabetically button, then it will be very frustrating, so we should take steps to prevent this.

**Reply by peppy** [reply]
I agree with these points. Allowing manual ordering but also having a "sort alphabetically" button is the best way forward.
